BlockchainSdk/BlockchainService/Aptos/AptosNetworkService.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  AptosNetworkService.swift
+//  BlockchainSdk
+//
+//  Created by skibinalexander on 29.01.2024.
+//  Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftyJSON
+
+class AptosNetworkService: MultiNetworkProvider {
+    // MARK: - Protperties
+    
+    let providers: [AptosNetworkProvider]
+    let blockchainDecimalValue: Decimal
+    
+    var currentProviderIndex: Int = 0
+    
+    // MARK: - Init
+    
+    init(providers: [AptosNetworkProvider], blockchainDecimalValue: Decimal) {
+        self.providers = providers
+        self.blockchainDecimalValue = blockchainDecimalValue
+    }
+    
+    // MARK: - Implementation
+    
+    func getAccount(address: String) -> AnyPublisher<AptosAccountInfo, Error> {
+        providerPublisher { provider in
+            provider
+                .getAccountResources(address: address)
+                .withWeakCaptureOf(self)
+                .tryMap { service, response in
+                    guard
+                        let accountJson = response.first(where: { $0.type == Constants.accountKeyPrefix }),
+                        let coinJson = response.first(where: { $0.type == Constants.coinStoreKeyPrefix })
+                    else {
+                        throw WalletError.failedToParseNetworkResponse
+                    }
+                    
+                    if let balanceValue = Decimal(coinJson.data.coin?.value), let sequenceNumber = Decimal(accountJson.data.sequenceNumber) {
+                        let decimalBalanceValue = balanceValue / service.blockchainDecimalValue
+                        return AptosAccountInfo(sequenceNumber: sequenceNumber.int64Value, balance: decimalBalanceValue)
+                    } else {
+                        throw WalletError.failedToParseNetworkResponse
+                    }
+                }
+                .eraseToAnyPublisher()
+        }
+    }
+    
+    func getGasUnitPrice() -> AnyPublisher<UInt64, Error> {
+        providerPublisher { provider in
+            provider
+                .getGasUnitPrice()
+                .withWeakCaptureOf(self)
+                .map { service, response in
+                    return response.gasEstimate
+                }
```

Comment:
`withWeakCaptureOf` лишнее

```suggestion
                .map { response in
                    return response.gasEstimate
                }
```